bokehjs/src/coffee/models/tools/actions/save_tool.coffee

```diff
@@ -0,0 +1,29 @@
+_ = require "underscore"
+
+ActionTool = require "./action_tool"
+
+class SaveToolView extends ActionTool.View
+
+  do: () ->
+    canvas = @plot_view.canvas_view.canvas[0]
+    name = "bokeh_plot.png"
+
+    if canvas.msToBlob?
+      blob = canvas.msToBlob()
+      window.navigator.msSaveBlob(blob, name)
+    else
+      link = document.createElement('a')
+      link.href = canvas.toDataURL('image/png')
+      link.download = name
+      link.dispatchEvent(new MouseEvent('click'))
```

Comment:
What about adding an on_click callback that removes the link element so that this cleans up after itself.